proto/message_contents/signature.proto

```diff
@@ -22,8 +22,16 @@ message Signature {
         bytes bytes = 1; // compact representation [ R || S ], 64 bytes
         uint32 recovery = 2; // recovery bit
     }
+    // Self-describing signature format meant for maximizing interoperability
+    message Generic {
+        bytes signature_bytes = 1; // signature bytes
+        bytes message_bytes = 2;
+        string algorithm = 3; // signature algorithm
```

Comment:
Can you provide an example of values that would belong here?
Particularly interested in scope

Reply:
If we carry the signed message payload without context, it'll be easy to accidentally do something along the lines of:
```
is_ok = public_key.verify(proto.message_bytes, proto.signature_bytes)
```
in our code unless we're very clear about what context the signature should be used for. This happens implicitly now because we need to regenerate the message using out-of-band context during signature verification.